pkg/driver/controller.go

```diff
@@ -0,0 +1,337 @@
+	if err != nil {
+		d.gidAllocator.releaseGid(accessPointsOptions.FileSystemId, gid)
+		if err == cloud.ErrAccessDenied {
+			return nil, status.Errorf(codes.Unauthenticated, "Access Denied. Please ensure you have the right AWS permissions: %v", err)
+		}
+		if err == cloud.ErrAlreadyExists {
+			return nil, status.Errorf(codes.AlreadyExists, "Access Point already exists")
+		}
+		return nil, status.Errorf(codes.Internal, "Failed to create Access point in File System %v : %v", accessPointsOptions.FileSystemId, err)
+	}
+
+	return &csi.CreateVolumeResponse{
+		Volume: &csi.Volume{
+			CapacityBytes: volSize,
+			VolumeId:      accessPointsOptions.FileSystemId + "::" + accessPointId.AccessPointId,
+			VolumeContext: map[string]string{},
+		},
+	}, nil
+}
+
+func (d *Driver) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequest) (*csi.DeleteVolumeResponse, error) {
+	klog.V(4).Infof("DeleteVolume: called with args %+v", *req)
+	volId := req.GetVolumeId()
+	if volId == "" {
+		return nil, status.Error(codes.InvalidArgument, "Volume ID not provided")
+	}
+
+	fileSystemId, _, accessPointId, err := parseVolumeId(volId)
+	if err != nil {
+		//Returning success for an invalid volume ID. See here - https://github.com/kubernetes-csi/csi-test/blame/5deb83d58fea909b2895731d43e32400380aae3c/pkg/sanity/controller.go#L733
+		klog.V(5).Infof("DeleteVolume: Failed to parse volumeID: %v, err: %v, returning success", volId, err)
+		return &csi.DeleteVolumeResponse{}, nil
+	}
+
+	//TODO: Add Delete File System when FS provisioning is implemented
+	if accessPointId != "" {
+
+		// Delete access point root directory if delete-access-point-root-dir is set.
+		if d.deleteAccessPointRootDir {
+			// Check if Access point exists.
+			// If access point exists, retrieve its root directory and delete it/
+			accessPoint, err := d.cloud.DescribeAccessPoint(ctx, accessPointId)
+			if err != nil {
+				if err == cloud.ErrAccessDenied {
+					return nil, status.Errorf(codes.Unauthenticated, "Access Denied. Please ensure you have the right AWS permissions: %v", err)
+				}
+				if err == cloud.ErrNotFound {
+					klog.V(5).Infof("DeleteVolume: Access Point %v not found, returning success", accessPointId)
+					return &csi.DeleteVolumeResponse{}, nil
+				}
+				return nil, status.Errorf(codes.Internal, "Could not get describe Access Point: %v , error: %v", accessPointId, err)
+			}
+
+			//Mount File System at it root and delete access point root directory
+			mountOptions := []string{"tls"}
+			target := TempMountPathPrefix + "/" + accessPointId
+			if err := d.mounter.MakeDir(target); err != nil {
+				return nil, status.Errorf(codes.Internal, "Could not create dir %q: %v", target, err)
+			}
+			if err := d.mounter.Mount(fileSystemId, target, "efs", mountOptions); err != nil {
+				os.Remove(target)
+				return nil, status.Errorf(codes.Internal, "Could not mount %q at %q: %v", fileSystemId, target, err)
+			}
+			os.RemoveAll(target + accessPoint.AccessPointRootDir)
```

Comment:
Should this error be checked for fine to ignore? What happens if the dir has been deleted but access point still exists? Say if the controller gets killed at line 247

Reply:
It should be fine, the controller will proceed to delete access point once it is restarted - [Os.RemoveAll](https://golang.org/pkg/os/#RemoveAll) returns nil if the path doesn't exist.

Reply:
ok, but we don't check if the err is nil.